tendermint/src/block/header.rs

```diff
@@ -139,10 +142,20 @@ pub struct Version {
     pub block: u64,
 
     /// App version
-    #[serde(with = "serializers::from_str")]
+    #[serde(
+        with = "serializers::from_str",
+        default = "Version::default_app_version"
+    )]
     pub app: u64,
 }
 
+impl Version {
```

Comment:
Can you please implement the default trait instead?

Reply:
I'm not sure how implementing the `Default` trait would solve this issue? You'd actually need to implement `Default` for the `u64` type, because serde calls `Default::default()` for the field whose value was omitted in the JSON, not the overall struct, as far as I understand it.